Source/WebCore/rendering/GridTrackSizingAlgorithm.cpp

```diff
@@ -1436,9 +1436,15 @@ bool GridTrackSizingAlgorithm::shouldExcludeGridItemForMasonryTrackSizing(const
     if (GridPositionsResolver::resolveGridPositionsFromStyle(*m_renderGrid, gridItem, m_direction).isIndefinite())
         shouldExcludeGridItemForMasonryTrackSizing = false;
 
-    // If the item is going past the end of track do not consider it for inclusion.
-    if (itemSpan.integerSpan() + trackIndex > tracks(m_direction).size())
-        shouldExcludeGridItemForMasonryTrackSizing = true;
+    // If the item is going past the end of track, check to see if we can shift the span to include it.
+    if (itemSpan.integerSpan() + trackIndex > tracks(m_direction).size()) {
+        auto shift = itemSpan.integerSpan() + trackIndex - tracks(m_direction).size();
+
+        if (shift > itemSpan.startLine())
+            shouldExcludeGridItemForMasonryTrackSizing = true;
+        else
+            itemSpan = GridSpan::translatedDefiniteGridSpan(itemSpan.startLine() - shift, itemSpan.integerSpan() + trackIndex - shift);
+    }
```

Comment:
I feel like this is not the right place to add this type of logic. Based off the current state of the function it takes in some input/constraints and answers the question of whether or not an item should be included for track sizing. I don't think we should be modifying those inputs and should instead have that be the responsibility of the caller if possible since they probably have more context as to what is going on

Reply:
Fixed, moved it up to be handled by the caller.